src/constructorio-id.js

```diff
@@ -32,6 +32,12 @@
 
         if (this.client_id_storage_location === 'cookie') {
           persisted_id = this.get_cookie(this.cookie_name_client_id);
+
+          if (persisted_id) {
+            // Persist `clientId` in cookie storage to ensure update of expiry date
+            this.delete_cookie(this.cookie_name_client_id);
```

Comment:
Apologies in advance for the wall of text! 😅 

I think we may want to look into this a bit deeper. I went down a rabbit hole while looking at some stuff for Gap and I realized that the identity module might not deleting cookies properly on beacons where the `cookie_domain` is specified. This is what the cookie deletion code does today:

```
document.cookie = "ConstructorioID_client_id=; expires=Thu, 01 Jan 1970 00:00:01 GMT;"
```

I believe this worked fine in the past when we didn't have the ability to specify `cookie_domain`, but if you try to run the code today, the deletion code might not be working the way we were expecting.

By default, when you call `document.cookie` without specifying a cookie domain, it will use the current domain that the code is running on. So if we're on `www.test.com`, that will be used as the domain. In our beacons we specify `.test.com` as the `cookie_domain`, which allows it to be surfaced across subdomains. However, when the identity module tries to delete it (using the code above), it defaults the domain to `www.test.com` which results in no cookie deleted since the cookie is set on `.test.com`.

If there is a `cookie_domain` specified, I think we should include that as part of `delete_cookie` wherever the function is called:

`delete_cookie(this.cookie_name_client_id, cookie_domain)`

The updated delete function would probably look something like this:

```
ConstructorioID.prototype.delete_cookie = function (name, domain) {
    let cookieValue = `${name}=;expires=Thu, 01 Jan 1970 00:00:01 GMT;`;
    
    if (domain) {
        cookieValue += `domain=${domain};`;
    }
    
    document.cookie = cookieValue;
};
```

My brain is kind of fried at this point, but happy to talk through it and explain further.

P.S. I could also just be crazy and typed this all out for no reason cause it seems to remove it just fine in the tests when using a cookie domain like `.localhost.com`. Also tried utilizing the cookie jar feature as well and it worked too. 😅

Reply:
Chatted through this w/ @esezen and this shouldn't be an issue. The case where we need to delete a cookie for a specific `cookie_domain` will be rare and in that situation we can handle them on an individual basis like we did with a recent apparel customer. 👍